src/parse.ts

```diff
@@ -50,24 +50,45 @@ export function parse(query: string, options?: JSONQueryParseOptions): JSONQuery
     return first
   }
 
-  const parseOperator = () => {
-    const left = parseParenthesis()
+  const parseOperator = (precedenceLevel: number) => {
+    const currentOperators = allOperators[precedenceLevel]
+    if (!currentOperators) {
+      return parseParenthesis()
+    }
+
+    let left = parseOperator(precedenceLevel + 1)
 
     skipWhitespace()
 
+    while (true) {
+      const name = parseOperatorName(currentOperators)
+      if (!name) {
+        break
+      }
+
+      const right = parseOperator(precedenceLevel + 1)
+      left = [name, left, right]
+    }
+
+    return left
+  }
+
+  const parseOperatorName = (currentOperators: Record<string, string>): string | undefined => {
     // we sort the operators from longest to shortest, so we first handle "<=" and next "<"
+    const sortedOperatorNames = Object.keys(currentOperators).sort((a, b) => b.length - a.length)
```

Comment:
Does it need to be re-sorted each time the function is called or could it be cached? (I see this function being called inside a while loop)

Reply:
You're right, this is not the most performant solution, I tried to keep the code as concise as possible. I do not think this will give performance issues since it's just the parsing step of the query and the list of operator names contains max 6 items. Unless there is a faster solution with about the same amount of code I think it's best to leave it as it is for now.